### PR TITLE
26

### LINKED ISSUE #26: InFile.java:37-38: Test Thread safety We need to...
The puzzle `22-576a91fb` from #22 has to be resolved: 

https://github.com/eo-cqrs/xfake/blob/50f7c778377fc4907470e2bfbeab2eb542e40188/src/main/java/io/github/eocqrs/xfake/InFile.java#L37-L38

The puzzle was created by @rultor on 26-May-23. 

Estimate: 45 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by l3r8yJ**
closes #26 

@h1alexbel slightly reorganized the library, wdyt

---

## PR-Codex overview
This PR removes a class and test, and modifies three others. It removes the `ImmutableReentrantLock` and its test, and modifies `InFile`, `FkStorage` and `Synchronized` classes. The changes consist of removing deprecated methods, adding thread safety and improving code readability.

### Detailed summary
- Remove `ImmutableReentrantLock` and its test
- Remove `lock` and `unlock` methods from `FkStorage`
- Add `Synchronized` decorator for `FkStorage`
- Improve thread safety in `InFile` and `Synchronized`
- Remove deprecated `XMLDocument` constructor
- Improve code readability

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by zoeself** [comment]
@l3r8yJ thank you for your Pull Request. I'll assign someone to review it soon. 

 If this PR solves a ``todo`` from the code, please don't forget to remove it.

**Comment by zoeself** [comment]
@h1alexbel please review this Pull Request. Deadline (when it should be merged or closed) is ``2023-05-29T18:02:20.353224``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

**Comment by l3r8yJ** [comment]
@h1alexbel review-ready

File: src/main/java/io/github/eocqrs/xfake/FkSynchronizedStorage.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * This is synchronized decorator for {@link FkStorage}.
+ * <b>This class is thread-safe.</b>
+ *
+ * @since 0.0.2
+ * @author l3r8yJ
+ */
+@RequiredArgsConstructor
+public final class FkSynchronizedStorage implements FkStorage {
```
Comment (h1alexbel) [inline src/main/java/io/github/eocqrs/xfake/FkSynchronizedStorage.java:20]: @l3r8yJ maybe simplify to just `Synchronized`?

WDYT?

Comment (l3r8yJ) [reply src/main/java/io/github/eocqrs/xfake/FkSynchronizedStorage.java:20]: @h1alexbel I'm not sure that Java would accept such a name

**Comment by l3r8yJ** [comment]
@h1alexbel fixed

**Review by h1alexbel — APPROVED** [review]
lgtm

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by h1alexbel** [comment]
@l3r8yJ thanks

**Comment by zoeself** [comment]
@h1alexbel thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**